### PR TITLE
PLAT-130491: SliderButton: Added 5-way navigation

### PR DESCRIPTION
**Pull request by daniel-stoian-lgp**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [x] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [x] Documentation was verified or is not changed
* [x] UI test was passed or is not needed
* [x] Screenshot test was verified or is not needed

* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
- agate/SliderButton could not be moved by direction keys. In addition, it is needed to verify a11y features via 5-way 
- agate/SliderButton focus state was not visually reflected 

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
- added handler to change value on arrow left/right key down
- added dedicated styling for focus state

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
PLAT-130491

### Comments

Enact-DCO-1.0-Signed-off-by: Daniel Stoian daniel.stoian@lgepartner.com

### CODE REVIEW

**Review by adrian-cocoara-lgp — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ The following is a curated list of changes in the Enact agate module, newest cha
 - `agate/DateTimePicker` prop `dayAriaLabel`, `hourAriaLabel`, `meridiemAriaLabel`, `minuteAriaLabel`, `monthAriaLabel`, and `yearAriaLabel`
 - `agate/Heading` prop `marqueOn` to determine when marquee begins
 - `agate/Popup` content padding and margin with `closeButton`
+- `agate/SliderButton` 5-way navigation functionality to change value via direction keys.
```
Comment (jeonghee27) [inline CHANGELOG.md:15]: I am not sure "added" is proper . Please decide it when merge.

File: SliderButton/tests/SliderButton-specs.js [hunk 2]
```diff
@@ -81,4 +81,34 @@ describe('SliderButton Specs', () => {
...
+		);
+
+		sliderButton.find('SliderButton').simulate('keyDown', {which: 39, keyCode: 39, code: 39});
+
+		const expected = 1;
+		const actual = handleChange.mock.calls[0][0].value;
+		expect(actual).toBe(expected);
+	});
+
+	test('should change value on arrow right left  press', () => {
```
Comment (jeonghee27) [inline SliderButton/tests/SliderButton-specs.js:100]:
```suggestion
	test('should change value on arrow right left press', () => {
```

File: SliderButton/tests/SliderButton-specs.js [hunk 3]
```diff
@@ -81,4 +81,34 @@ describe('SliderButton Specs', () => {
...
+
+	test('should change value on arrow right key press', () => {
+		const handleChange = jest.fn();
+		const sliderButton = mount(
+			<SliderButton onChange={handleChange}>
+				{children}
+			</SliderButton>
+		);
+
+		sliderButton.find('SliderButton').simulate('keyDown', {which: 39, keyCode: 39, code: 39});
```
Comment (jeonghee27) [inline SliderButton/tests/SliderButton-specs.js:93]: I think only "keyCode" is sufficient. Strangely, only Input used it like this. I don't know if there is a reason

File: tests/ui/specs/SliderButton/SliderButton-specs.js [hunk 4]
```diff
@@ -23,6 +23,17 @@ describe('SliderButton', function () {
 
 			expect(sliderButton.getKnob().getAttribute('proportion')).to.equal('1');
 		});
+
+		describe('5-way', function () {
+			it('should move focus between options on 5-way navigation', function () {
```
Comment (jeonghee27) [inline tests/ui/specs/SliderButton/SliderButton-specs.js:28]: Isn't it better to "change value" than "move focus"?

File: CHANGELOG.md (hunk 1 above)
Comment (jeonghee27) [reply CHANGELOG.md:15]: Please move to "Changed"

File: SliderButton/tests/SliderButton-specs.js (hunk 3 above)
Comment (daniel-stoian-lgp) [reply SliderButton/tests/SliderButton-specs.js:93]: 👍

**Review by jeonghee27 — APPROVED** [review]
LGTM

File: SliderButton/tests/SliderButton-specs.js [hunk 5]
```diff
@@ -81,4 +81,34 @@ describe('SliderButton Specs', () => {
...
+		);
+
+		sliderButton.find('SliderButton').simulate('keyDown', {keyCode: 39});
+
+		const expected = 1;
+		const actual = handleChange.mock.calls[0][0].value;
+		expect(actual).toBe(expected);
+	});
+
+	test('should change value on arrow right left press', () => {
```
Comment (MikyungKim) [inline SliderButton/tests/SliderButton-specs.js:100]: on arrow left press?

Comment (daniel-stoian-lgp) [reply SliderButton/tests/SliderButton-specs.js:100]: right :D

**Review by MikyungKim — COMMENTED** [review]
The Carbon skin, when the SliderButton got focused, it looks like
image
I found it's reproducible on develop too. WDYT? Should it work like this?

**Comment by daniel-stoian-lgp** [comment]
> The Carbon skin, when the SliderButton got focused, it looks like
> image
> I found it's reproducible on develop too. WDYT? Should it work like this?

Seems to be a design issue. Slider focus-bg-color is the same as the knob color. Should we use a different color for one of them? I don't know the guide for Carbon skin

**Comment by MikyungKim** [comment]
> > The Carbon skin, when the SliderButton got focused, it looks like
> > image
> > I found it's reproducible on develop too. WDYT? Should it work like this?
> 
> Seems to be an older design issue. Slider focus-bg-color is the same as the slider knob color. Should we use a different color for one of them? I don't know the guide for Carbon skin

🤔 Let's skip for now.

**Review by MikyungKim — APPROVED** [review]
LGTM

**Pull request merged**